src/core/jit/exec.rs

```diff
@@ -6,19 +6,38 @@ use derive_getters::Getters;
 use futures_util::future::join_all;
 
 use super::context::Context;
-use super::{DataPath, Field, Nested, OperationPlan, Request, Response, Store};
+use super::{DataPath, Field, LocationError, Nested, OperationPlan, Request, Response, Store};
 use crate::core::ir::model::IR;
-use crate::core::jit;
 use crate::core::jit::synth::Synth;
+use crate::core::jit::{self};
 use crate::core::json::JsonLike;
 
 type SharedStore<Output, Error> = Arc<Mutex<Store<Result<Output, Error>>>>;
 
+#[derive(Debug, Clone, Getters)]
+pub struct ExecutionEnv<Input> {
+    plan: OperationPlan<Input>,
+    err: Arc<Mutex<Vec<LocationError<jit::error::Error>>>>,
+}
+
+impl<Input> ExecutionEnv<Input> {
+    pub fn new(plan: OperationPlan<Input>) -> Self {
+        Self { plan, err: Arc::new(Mutex::new(vec![])) }
+    }
+    pub fn add_error(&self, new_error: LocationError<jit::error::Error>) {
+        self.err.lock().unwrap().push(new_error);
+    }
+
+    pub fn get_errors(&self) -> Vec<LocationError<jit::error::Error>> {
+        self.err.lock().unwrap().clone()
```

Comment:
don't do clone here, better let caller decide what to do

Reply:
done